### ISSUE TITLE
Add error param to input

### ISSUE THREAD

**Issue by penx** [alphagov/govuk-frontend#1150]
At the moment, date-input passes a CSS class to input when it wants to mark a field as having an error:

https://github.com/alphagov/govuk-frontend/blob/e70f59d15a3cab9c949ef8f558d1eb49be7e4c91/src/components/date-input/template.njk#L64

https://github.com/alphagov/govuk-frontend/blob/e70f59d15a3cab9c949ef8f558d1eb49be7e4c91/src/components/date-input/date-input.yaml#L125-L126

This implies that the user of the date-input has inferred knowledge of the inner workings (CSS class name) of input.

I propose that an 'error' param is added to input so that, when using the input template, the user (be it an application using the input template or another component) doesn't need to have inferred knowledge of the class names. The classes param would still exist so this approach would also still work, but the preferred approach would be to use the error param.

This would mean adding an error param to input as follows:

before:
```nunjucks
  <input class="govuk-input {%- if params.classes %} {{ params.classes }}{% endif %} {%- if params.errorMessage %} govuk-input--error{% endif %}" id="{{ params.id }}" name="{{ params.name }}" type="{{ params.type | default('text') }}"
```

after:
```nunjucks
  <input class="govuk-input {%- if params.classes %} {{ params.classes }}{% endif %} {%- if params.errorMessage | params.error %} govuk-input--error{% endif %}" id="{{ params.id }}" name="{{ params.name }}" type="{{ params.type | default('text') }}"
```

And changing the date-input examples as follows:

before:
```yaml
        name: day
        classes: govuk-input--width-2 govuk-input--error
```

after:
```yaml
        name: day
        classes: govuk-input--width-2
        error: true
```

This may seem like a trivial issue for users of the nunjucks templates - but when working on ports (e.g. using CSSinJS or CSS modules) it is useful for one component/template to not have to have knowledge of the inner workings of another.

**Comment by nickcolley** [comment]
I believe this is the same as this issue we closed: https://github.com/alphagov/govuk-frontend/issues/460

There has been some recent discussion internally about this since the GOV.UK Publishing team is very keen on avoiding using classes for this same reason.

**Comment by penx** [comment]
Yep looks the same. Can #460 be reopened?

**Comment by aliuk2012** [comment]
Closing this as it is part of #460

**Comment by penx** [comment]
Can this be reopened as it doesn't seem to have been addressed in #460?

([as per my comment](https://github.com/alphagov/govuk-frontend/issues/460#issuecomment-953924321), for this to be resolved, the fixtures should not contain CSS classes)
